### DISCUSSION TITLE
SSR getInitialProps子组件使用

### DISCUSSION THREAD

**Discussion by zfhslfgit** [umijs/umi#6750] [Q&A] [answered]
页面有很多公共组件，里面的接口请求也需要走SSR
但是getInitialProps在子组件无法使用

**Comment by robinv8** [comment, marked as answer]
如果在子组件使用 getInitialProps 会加大 SSR 的复杂度，不建议这么做，也没必要这么做

**Reply by xy1041** [reply]
那么当同一个公共组件（如nav）中的某些数据（比如nav的数量和内容是可配置的）需要通过getInitialProps获得一些服务端数据渲染之后直接返回的情况，如何收束各个页面中的公共组件数据来源呢，CSR的话可以通过在入口js或者nav的第一次载入的时机获得数据并存储到redux中，之后页面跳转时可以直接从redux中保证获得数据，整个过程无需依赖上层页面组件的参数传入，但是ssr的情况下useEffect无法触发，在通用的运行时配置（即app.tsx）里进行请求的话却不能像getInitialProps一样把返回的数据用window.g_initialData的方式传递到浏览器上，有什么办法可以让在多个组件里使用，需要的数据则是根据域名请求的组件在不依赖最外层的页面组件传入的情况下进行SSR呢

**Reply by robinv8** [reply]
不好意思来迟了，针对这个问题我已在 https://segmentfault.com/q/1010000040447548 做了解答
